include/swift/AST/Module.h

```diff
@@ -802,6 +816,13 @@ class ModuleDecl : public DeclContext, public TypeDecl {
                      unsigned Alignment = alignof(ModuleDecl));
 };
 
+/// required to put an ImportedModule in a Set
+inline bool operator==(const ModuleDecl::_ImportedModule LHS, const ModuleDecl::_ImportedModule RHS) {
```

Comment:
If you look at other places in the code, we typically define the `operator==` overload inside the type itself like

```
bool operator==(const Typename &Other) const { /* ... */ }
```

and marking it `friend` if necessary. Could you do the same here, moving the definition of the overload inside the definition of `ImportedModule`?

Reply:
Resolved in af9a745. I had a syntax problem in an earlier commit where things would not work unless the operator was outside, but I rewrote it correctly inside the type and now it works.